api/v1alpha1/linodecluster_webhook.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2023 Akamai Technologies, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"slices"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	. "github.com/linode/cluster-api-provider-linode/clients"
+)
+
+// log is for logging in this package.
+var linodeclusterlog = logf.Log.WithName("linodecluster-resource")
+
+// SetupWebhookWithManager will setup the manager to manage the webhooks
+func (r *LinodeCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// TODO(user): EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+
+// TODO(user): change verbs to "verbs=create;update;delete" if you want to enable updation and deletion validation.
+//+kubebuilder:webhook:path=/validate-infrastructure-cluster-x-k8s-io-v1alpha1-linodecluster,mutating=false,failurePolicy=fail,sideEffects=None,groups=infrastructure.cluster.x-k8s.io,resources=linodeclusters,verbs=create,versions=v1alpha1,name=vlinodecluster.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &LinodeCluster{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *LinodeCluster) ValidateCreate() (admission.Warnings, error) {
+	linodeclusterlog.Info("validate create", "name", r.Name)
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultWebhookTimeout)
+	defer cancel()
+
+	return nil, r.validateLinodeCluster(ctx, &defaultLinodeClient)
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *LinodeCluster) ValidateUpdate(old runtime.Object) (admission.Warnings, error) {
+	linodeclusterlog.Info("validate update", "name", r.Name)
+
+	// TODO(user): fill in your validation logic upon object update.
+	return nil, nil
```

Comment:
Since we allow the region to be updated, we should probably change this to 
```
return nil, r.validateLinodeCluster(ctx, &defaultLinodeClient)
```

Reply:
Before we consider these changes, do you know if we have a story for moving clusters to a different region as of now? 

It looks the `spec.region` field in the LinodeMachines themselves immutable and the LinodeCluster controller doesn't even reconcile updates.

Reply:
AFAIK there's no ask to support that... *yet*

Reply:
> maybe make it immutable since the VPC would need to have its region also updated for changing the LinodeCluster region to work as expected when using a VPC

Going through the specifications again, I think this option would be the best at the moment since:
1. The `.spec.Region` of a LinodeVPC is immutable
2. The LinodeCluster controller currently doesn't reconcile `.spec.Region` updates

I think if we do decide to support this feature, we can enable update validation for LinodeClusters then.